File: order-service/src/main/java/com/paulopacifico/orderservice/messaging/api/InventoryReservedEvent.java
package com.paulopacifico.orderservice.messaging.api;

import java.time.OffsetDateTime;
import java.util.UUID;

public record InventoryReservedEvent(
        UUID eventId,
        Long orderId,
        String orderNumber,
        String skuCode,
        Integer reservedQuantity,
        OffsetDateTime occurredAt
) {
}

File: inventory-service/src/main/kotlin/com/paulopacifico/inventoryservice/messaging/api/InventoryReservedEvent.kt
package com.paulopacifico.inventoryservice.messaging.api

import java.time.OffsetDateTime
import java.util.UUID

data class InventoryReservedEvent(
    val eventId: UUID,
    val orderId: Long,
    val orderNumber: String,
    val skuCode: String,
    val reservedQuantity: Int,
    val occurredAt: OffsetDateTime,
)
